src/dehb/utils/config_repository.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import numpy as np
+
+
+@dataclass
+class ConfigItem:
+    config_id: int
+    config: np.array
+    fidelities: dict
+
+@dataclass
+class ResultItem:
+    score: float
+    cost: float
+    info: dict
+
+class ConfigRepository:
+    def __init__(self) -> None:
+        self.configs = []
+
+    def reset(self) -> None:
+        self.configs = []
+
+    def announce_config(self, config: np.array, fidelity: float) -> int:
+        config_id = len(self.configs)
```

Comment:
A docstring explainer would be good here, explaining the data structure so to say.